core/src/main/java/org/kiwix/kiwixmobile/core/main/FileReader.kt

```diff
@@ -15,31 +15,26 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.core.main;
+package org.kiwix.kiwixmobile.core.main
 
-import android.content.Context;
-import java.io.BufferedReader;
-import java.io.InputStream;
-import java.io.InputStreamReader;
+import android.content.Context
+import java.io.IOException
 
-public class FileReader {
-
-  public String readFile(String filePath, Context context) {
-    try {
-      StringBuilder buf = new StringBuilder();
-      InputStream json = context.getAssets().open(filePath);
-      BufferedReader in =
-        new BufferedReader(new InputStreamReader(json, "UTF-8"));
-      String str;
-
-      while ((str = in.readLine()) != null) {
-        buf.append(str);
+class FileReader {
+  fun readFile(filePath: String?, context: Context): String {
+    return try {
+      val buf = StringBuilder()
+      val json = context.assets.open(filePath)
+      val reader = json.bufferedReader()
```

Comment:
you were much closer with the `use` method! explore more bufferedReader methods  with `ctrl+space`

Or I can tell you what to do precisely but that lessens the learning experience

Reply:
<img width="1440" alt="Screenshot 2020-08-20 at 3 57 54 PM" src="https://user-images.githubusercontent.com/43576162/90759344-ea13f080-e2fd-11ea-9db0-5b44fc6328a3.png">
Is this the correct way?

Reply:
That is a way and would produce a correct result but we can do this without a loop and even without a StringBuilder.

`use` is close but not the precise method

Reply:
<img width="1440" alt="Screenshot 2020-08-20 at 4 10 55 PM" src="https://user-images.githubusercontent.com/43576162/90760584-bb971500-e2ff-11ea-9f38-042f0d58f5a5.png">
@macgills how about now?

Reply:
better than what I was thinking of! My mind went to `useLines{ it.joinToString{it} }`

Have you tested the table of contents is still working?

Reply:
@macgills I am new to this project. I don't know how it should work.

Reply:
So open the app, download a book, open a book, there is a button on the toolbar to open the right hand side panel which shows a table of contents which can be clicked to jump to a section, it can also be swiped open

Reply:
yes @macgills it works fine.

Reply:
then commit away